libpd_wrapper/x_libpdreceive.c

```diff
@@ -19,33 +18,39 @@ static t_class *libpdrec_class;
 typedef struct _libpdrec {
   t_object x_obj;
   t_symbol *x_sym;
+  t_libpdhooks *x_hooks;
 } t_libpdrec;
 
 static void libpdrecbang(t_libpdrec *x) {
-  if (libpd_banghook) (*libpd_banghook)(x->x_sym->s_name);
+  if (x->x_hooks->h_banghook)
+    (*x->x_hooks->h_banghook)(x->x_sym->s_name);
 }
 
 static void libpdrecfloat(t_libpdrec *x, t_float f) {
-  if (libpd_floathook) (*libpd_floathook)(x->x_sym->s_name, f);
-  if (libpd_doublehook) (*libpd_doublehook)(x->x_sym->s_name, f);
+  if (x->x_hooks->h_floathook)
+    (*x->x_hooks->h_floathook)(x->x_sym->s_name, f);
+  if (x->x_hooks->h_doublehook)
```

Comment:
My understanding is that at most one of h_{float,double}hook will be non-null, so "else if" seems closer to the intention?

Reply:
Probably right. If one is set the other will be NULL. I'll update this in `z_queued.c` as well.